### DISCUSSION TITLE
Laravel Nova Action Logs show finished, even when jobs fail.

### DISCUSSION THREAD

**Discussion by kerbs-ns** [laravel/nova-issues#5292] [Q&A]
- Laravel Version: 9.49.0 
- Nova Version: 4.21.0
- PHP Version: 8.2.1

- Sail v1.19.0 

### Description:

- Laravel Job is created to use batch implements ShouldQueue and use Batchable
- Laravel Nova Action implements BatchableAction, ShouldQueue
- Model use Actionable

When queues are disabled/removed/not in use, nova action log does work as expected. 

When Nova Action is batchable and queueable and uses job that is batchable and queueable. All models do get nova activity log that are finished, even if they throw exception. 

Maybe this is not issue ?!

What is expected to happen when action is queueable, job is queueable and job create resource/model or there is exception. 

What should we see in activity log ?

**Comment by crynobone** [comment]
Action Log capture and log state of running the action on each model. While the whole job may be failed Nova may still have successfully run the action to some model and this will be reflected in the model action log. 

To review the job status you may need to look into Horizon or other tools.

**Reply by kerbs-ns** [reply]
I have to say that action does behave differently if is being used in queue or not, and that feels strange. Specially that admin can't know if end result is true or false.

in this world where if action is queued it is up to job to log some event in action.

What do you suggest and how to create activity log... when  NovaAction::for... is quite limited... for example you can't change name out of the box

Thanks in advanced.
